### PR TITLE
Model api

### PR DESCRIPTION
**Pull request by cfrancesco**
- factory module
- ckpt to hdf5 conversion
- ljspeech API

### CODE REVIEW

File: model/models.py [hunk 1]
```diff
@@ -512,3 +540,37 @@ def _make_min_duration_mask(self, encoded_text, phoneme_min_duration):
                 phon_idx = self.text_pipeline.tokenizer(item[0])[0]
                 new_mask[np_text == phon_idx] = item[1]
         return tf.cast(tf.convert_to_tensor(new_mask), tf.float32)
+    
+    def build_model_weights(self, path: str = None):
```
Comment (cschaefer26) [inline model/models.py:544]: Return types missing.

File: model/factory.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+def tts_ljspeech(version='v1') -> [tf.keras.models.Model, dict]:
+    config_name = f'ljspeech_tts_config_{version}.yaml'
+    weights_name = f'ljspeech_tts_weights_{version}.hdf5'
+    remote_dir = 'https://public-asai-dl-models.s3.eu-central-1.amazonaws.com/TransformerTTS/api_weights/'
+    config_path = tf.keras.utils.get_file(config_name, remote_dir + config_name)
+    weights_path = tf.keras.utils.get_file(weights_name, remote_dir + weights_name)
+    return tts_custom(config_path, weights_path)
+
+
+def tts_custom(config_path: str, weights_path: str) -> [tf.keras.models.Model, dict]:
```
Comment (cschaefer26) [inline model/factory.py:18]: I am pretty sure that type highlighting won't work with this (instead use Tuple[tf.keras.models.Model, dict].

File: model/models.py (hunk 1 above)
Comment (cfrancesco) [reply model/models.py:544]:
```suggestion
    def build_model_weights(self, path: str = None)-> None:
```

File: model/factory.py (hunk 2 above)
Comment (cfrancesco) [reply model/factory.py:18]: True.
fiy from python 3.9:
https://docs.python.org/3/whatsnew/3.9.html#type-hinting-generics-in-standard-collections

**Pull request merged**